### PR TITLE
:Fix memory leak in hnsw_gettuple

### PR DESCRIPTION
**Pull request by knizhnik**

### CODE REVIEW

**Comment by bayandin** [comment]
MacOS CI failures are not related, they will be fixed by https://github.com/neondatabase/pg_embedding/pull/45

File: embedding.c [hunk 1]
```diff
@@ -295,6 +295,7 @@ hnsw_gettuple(IndexScanDesc scan, ScanDirection dir)
 		int         n_items;
 		size_t      n_results;
 		label_t*    results;
+		bool        res;
```
Comment (save-buffer) [inline embedding.c:298]: I think a better name would be something like `search_succeeded`

**Review by save-buffer — APPROVED** [review]
Looks good, just would prefer to rename the variable to be more descriptive.

File: embedding.c (hunk 1 above)
Comment (knizhnik) [reply embedding.c]: Replaced

**Pull request merged**